community/values/src/main/java/org/neo4j/values/storable/DateTimeValue.java

```diff
@@ -504,24 +530,17 @@ int unsafeCompareTo( Value other )
         return cmp;
     }
 
-    private boolean areSameZone( ZoneId thisZone, ZoneId thatZone, boolean thisIsOffset, boolean thatIsOffset )
+    private int compareNamedZonesWithMapping( ZoneId thisZone, ZoneId thatZone )
     {
-        if ( thisIsOffset && thatIsOffset )
-        {
-            return thisZone.equals( thatZone );
-        }
-        else if ( !thisIsOffset && !thatIsOffset )
+        short indexThisZone = TimeZones.map( thisZone.getId() );
+        short indexThatZone = TimeZones.map( thatZone.getId() );
+        int cmp = Short.compare( indexThisZone, indexThatZone );
```

Comment:
The code here does not order by alphabetical order, but by the order in the internal mapping table.

If we wanted to order alphabetically (which we probably want, since that is stable across implementations), the only thing we might want to use the mapping table for is for _normalizing_ the zone ids - i.e. map to internal id and then map that internal id back to a string.